modules/fedimint-lnv2-server/src/lib.rs

```diff
@@ -499,6 +514,20 @@ impl ServerModule for Lightning {
                 )
                 .await;
 
+                let stream_index = dbtx
+                    .get_value(&IncomingContractStreamIndexKey)
+                    .await
+                    .unwrap_or(0);
+
+                dbtx.insert_entry(&IncomingContractStreamKey(stream_index), contract)
+                    .await;
+
+                dbtx.insert_entry(&IncomingContractIndexKey(outpoint), &stream_index)
+                    .await;
+
+                dbtx.insert_entry(&IncomingContractStreamIndexKey, &(stream_index + 1))
+                    .await;
+
```

Comment:
Some documentation would be good here for future readers. If I am understanding it correctly:

 - We have a mapping of `stream_index` (`u64`) -> `IncomingContract`
 - We have a global index of the current index (`IncomingContractStreamIndexKey`)
 - We have a mapping from `OutPoint` to `stream_index`. Allows for lookup of contract by `OutPoint`, while also being able to read sequentially using `IncomingContractStreamPrefix`